wallet/test/Test/Pos/Util/MnemonicsSpec.hs

```diff
@@ -0,0 +1,63 @@
+module Test.Pos.Util.MnemonicsSpec (spec, Entropy(..)) where
+
+import           Universum
+
+import           Data.ByteString.Char8 (pack)
+import           Data.Set (Set)
+import           Test.Hspec (Spec, describe, it, shouldSatisfy, xit)
+import           Test.Hspec.QuickCheck (modifyMaxSuccess)
+import           Test.QuickCheck (Arbitrary (..), property)
+import           Test.QuickCheck.Gen (Gen (..), oneof, vectorOf)
+import           Test.QuickCheck.Random (mkQCGen)
+
+import           Pos.Util.BackupPhrase (BackupPhrase (..), safeKeysFromPhrase)
+import           Pos.Util.Mnemonics (defMnemonic, fromMnemonic, toMnemonic)
+import           Pos.Wallet.Web.ClientTypes.Functions (encToCId)
+import           Pos.Wallet.Web.ClientTypes.Types (CId)
+
+import qualified Data.Set as Set
+
+
+spec :: Spec
+spec = do
+    describe "Pos.Util.Mnemonics" $ do
+        modifyMaxSuccess (const 10000) $ it "toMnemonic >=> fromMnemonic = Right" $ property $
+            \(Entropy ent) -> (toMnemonic ent >>= fromMnemonic) == Right ent
+
+        it "No example mnemonic" $
+            fromMnemonic defMnemonic `shouldSatisfy` isLeft
+
+        it "No empty mnemonic" $
+            (fromMnemonic "") `shouldSatisfy` isLeft
+
+        it "No empty entropy" $
+            (toMnemonic "") `shouldSatisfy` isLeft
+
+        xit "entropyToWalletId is injective " $
+            let
+                inject :: Ord b => (a -> b) -> [a] -> Set b
+                inject fn xs =
+                    foldl' (\acc x -> Set.insert (fn x) acc) Set.empty xs
+
+                genList :: Arbitrary a => Int -> [a]
+                genList n =
+                    unGen (vectorOf n arbitrary) (mkQCGen 14) 14
```

Comment:
hmm why is this not a property? You could write:

```haskell
prop "entorpyToWalletId is injective" 
    $ forAll (vectorOf 10000 arbitrary) 
    $ \inputs ->
        length (inject entropyToWalletId inputs) === length inputs
```

Reply:
I was a property initially, but I switched back to a simple one-off check. Conceptually, it's a bit of the same; we want to test whether `entropyToWalletId` is injective, so we want to check that the function doesn't map two inputs to a same output. Testing 1000 times on lists of size n<sub>i</sub>, is the same as testing one time on a list of size Σn<sub>i</sub>. 

We aren't really interested in the various shapes of the inputs here, but just want to have a lot of them, ideally, all of them if it was possible.

Reply:
Nevertheless... Having it as a quickcheck property gives us a progress feedback instead of just, hanging there for a minute.